applications/portal/frontend/src/components/home/ExplorationSpinalCordDialog.tsx

```diff
@@ -36,35 +27,106 @@ const useStyles = makeStyles(() => ({
       marginBottom: '0.75rem'
     },
   },
+  editorWrapper: {
+    '&:hover': {
+      borderColor: '#fff'
+    },
+    '&:focus-within': {
+      borderColor: '#7b61ff'
+    },
+  },
 }));
 
+const DESCRIPTION_KEY = "description";
+
 export const ExplorationSpinalCordDialog = (props: any) => {
+
+  const api = WorkspaceService.getApi();
   const classes = useStyles();
-  const { open, handleClose } = props;
+  const commonClasses = common();
+  const { open, handleClose, experiment, tagsOptions, onExperimentChange } = props;
+
+  const [name, setName] = React.useState<string>(experiment.name);
+  const [description, setDescription] = React.useState<string>(experiment.description);
+  const [initialTags, setInitialTags] = React.useState(experiment.tags.map((t: any) => t.name));
+  const [tags, setTags] = React.useState(experiment.tags.map((t: any) => t.name));
+  const [validationErrors, setValidationErrors] = React.useState(new Set([]));
+
+  const handleFormChange = (newValue: any, setState: any, errorKey: string) => {
+    validationErrors.delete(errorKey);
+    setValidationErrors(validationErrors);
+    setState(newValue);
+  };
+
+  const onTagsChange = (e: any, value: string[]) => {
+    setTags(value);
+  };
+
+  const deleteTag = async (tag: string) => {
+    try {
+      await api.deleteTagExperiment(experiment.id.toString(), tag);
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  const addTags = async (tags: string[]) => {
+    try {
+      await api.addTagsExperiment(experiment.id.toString(), tags);
+    } catch (error) {
+      console.error(error);
+    }
+  }
+
+  const handleAction = async () => {
+    const deletedTags = initialTags.filter((tag: any) => !tags.includes(tag));
+    const addedTags = tags.filter((tag: any) => !initialTags.includes(tag));
+
+    // API calls accordingly
+    if (deletedTags.length > 0) {
+      deletedTags.forEach((tag: any) => deleteTag(tag));
+    }
+    if (addedTags.length > 0) {
+      addTags(addedTags);
+    }
+    // update the experiment
+    const res = await api.updateExperiment(experiment.id.toString(), name, description);
+
+    onExperimentChange(res.data.id);
+    setInitialTags(tags);
+    handleClose();
+  };
 
   return (
     <Modal
       dialogActions={true}
-      actionText="Create"
+      actionText="Edit"
       disableGutter={true}
       open={open}
       handleClose={handleClose}
+      handleAction={handleAction}
       title="Exploration of the spinal cord"
```

Comment:
This title doesn't make much sense, can you check with @neuroBazzi or @syamace-metacell what they recommend to use?

Reply:
@afonsobspinto Based on the designs, the title is supposed to be the current experiment's name, the one that the user is currently editing it's info & tags. So in this case (the video above), "Dataset V3 - Edited" should be the title. cc @neuroBazzi